src/main/java/com/pinterest/secor/parser/AvroMessageParser.java

```diff
@@ -31,7 +32,23 @@ public long extractTimestampMillis(final Message message) {
         try {
             GenericRecord record = schemaRegistryClient.decodeMessage(message.getTopic(), message.getPayload());
             if (record != null) {
-                Object fieldValue = record.get(mConfig.getMessageTimestampName());
+                String delimiter = mConfig.getMessageTimestampNameSeparator();
+                String timestampName = mConfig.getMessageTimestampName();
+                Object fieldValue;
+                
+                if (delimiter != null && !delimiter.isEmpty()) {
+                  String separatorPattern = Pattern.quote(delimiter);
+                  String[] fieldArray = timestampName.split(separatorPattern);
+                  int nameArrayLength = fieldArray.length - 1;
+
+                  for(int i = 0; i < nameArrayLength; i++) {
+                    record = (GenericRecord) record.get(fieldArray[i]);
```

Comment:
Do you want to generate some log warnings if some of the fields are not present in the record?

Reply:
Might make sense in case timestampRequired is false. I will add it.